### ISSUE TITLE
why .add[...] methods does not accept ts.Node or Node<ts.Node>?

### ISSUE THREAD

**Issue by Jack-Works** [dsherret/ts-morph#222]
```typescript
import * as ts from 'typescript'
import * as ast from 'ts-simple-ast'
declare const statements: ts.Statement[]
declare const sourcefile: ast.SourceFile

sourcefile.addStatements(statements)
// That doesn't work

sourcefile.addStatements(statements.map(statment => ast.createWrappedNode(statment)))
// Doesn't work either
```

addStatements just accept `string`, but I already have nodes, why can't I just add them?

**Comment by Jack-Works** [comment]
Before, using typescript
```typescript
const statements = transformer.statements.map(renderToString).join('\n')
const declarations = transformer.declarations.map(renderToString).join('\n')
const result = (statements + '\n' + declarations)
const code = ts.createSourceFile('generated.ts', result, ts.ScriptTarget.Latest, false)
```

After use ts-simple-ast
```typescript
const render = new ast.default()
const statements = transformer.statements.map(renderToString).join('\n')
const declarations = transformer.declarations.map(renderToString).join('\n')
const result = (statements + '\n' + declarations)
const file = render.createSourceFile('generated.ts', result)
```

I thought ts-simple-ast can provide a more clear way like this, but there is no api like this, so code doesn't get simple...
```typescript
const file = render.createSourceFile('generated.ts')
file.addStatements(transformer.statements)
file.addDeclarations(transformer.declarations)
```

**Comment by dsherret** [comment]
@Jack-Works I think making these methods accept nodes would give a false impression of what's going on behind the scenes. Between each manipulation, the source file text itself is updated, then a new tree is parsed from that new source file text, and finally the existing wrapped nodes are back filled with the new updated node. This means that the nodes in the tree will always match the source file text (rather than if you manipulate the ast directly and the underlying text won't match... if they don't match then you will get all kinds of incorrect results from the underlying typescript nodes).

What I think would be good in this scenario is something sort of along the lines of what you were mentioning and that's to add a `printNode` utility function:

```ts
import Ast, {printNode} from "ts-simple-ast";

const render = new Ast();

const result = [...transformer.statements, ...transformer.declarations].map(printNode).join("\n");
const file = render.createSourceFile('generated.ts', result);
```

Would that work for you?

**Comment by dsherret** [comment]
The `printNode` method is now available in 5.6.0. You can read about it [here](https://dsherret.github.io/ts-simple-ast/utilities)

**To answer this question once again in other words...**

This library traverses the AST, but all the manipulations are done on the underlying source file text. With this library, the underlying AST is always in sync with the source file text at all times and the underlying AST is recreated from the source file text after each manipulation—it doesn't build up an AST gradually and then eventually print the AST to source file text.

If you really wish to use the compiler API to create a node, then I recommend building it up as you normally would using the compiler API and then use the `printNode` function from this library to convert it to text and then call a method to insert the text where you want to insert it (as shown in my previous comment's code).
